### ISSUE TITLE
inla.set.hyper failure: length > 1 in coercion to logical

### ISSUE THREAD

**Issue by bastistician** [hrue/r-inla#21]
Running R with strict `_R_CHECK_LENGTH_1_LOGIC2_=verbose`, I get a 

> failure: length > 1 in coercion to logical

from the following line:

https://github.com/hrue/r-inla/blob/aa76a9547d4085dfbaa7765349b9798233178a4b/rinla/R/hyper.R#L244

Debugging shows that in my example, `h[["param"]]` equals `c(0, 0.001)`, so is a vector of length 2.
I guess the above code should thus be replaced by something like

```r
test.val = !is.null(h) && !is.null(h[[key]]) && any(!is.na(h[[key]]))
```

**Comment by hrue** [comment]
you're right.

thanks!

H


On Fri, 2020-12-04 at 06:50 -0800, Sebastian Meyer wrote:
> Running R with strict _R_CHECK_LENGTH_1_LOGIC2_=verbose, I get a
> failure: length > 1 in coercion to logical
> from the following line:
> https://github.com/hrue/r-inla/blob/aa76a9547d4085dfbaa7765349b9798233178a4b/rinla/R/hyper.R#L244
> Debugging shows that in my example, h[["param"]] equals c(0, 0.001),
> so is a vector of length 2.
> I guess the above code should thus be replaced by something like
> test.val = !is.null(h) && !is.null(h[[key]]) && any(!is.na(h[[key]]))
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

-- 
Håvard Rue
Professor of Statistics
Statistics Program, CEMSE Division
King Abdullah University of Science and Technology
Thuwal 23955-6900, Saudi Arabia

Email: haavard.rue@kaust.edu.sa
Office: +966 (0)12 808 0640  
Mobile: +966 (0)54 470 0421
Research group: bayescomp.kaust.edu.sa  
R-INLA project: www.r-inla.org
Zoom: kaust.zoom.us/my/haavard.rue

**Comment by bastistician** [comment]
Fixed in 2e16ac9b2bbb96a660cc37d82287498c29066bbd.
